### PR TITLE
Making smsutil as class.

### PR DESCRIPTION
**Pull request by ayush03agarwal**

### CODE REVIEW

**Comment by kingster** [comment]
Why would a util be a class? and not a singleton object?

**Comment by ayush03agarwal** [comment]
CharsetEncoder is not thread safe and gives java.lang.IllegalStateException when multiple request hits canEncode method simultaneously .

**Comment by ayush03agarwal** [comment]
let me verify once again

**Comment by kingster** [comment]
Yes, its not safe as per
https://docs.oracle.com/javase/6/docs/api/java/nio/charset/CharsetEncoder.html

You can probably use thread-local variables, instead of creating entire
util class everytime.

Thanks
Kinshuk

**Comment by ayush03agarwal** [comment]
Made SmsUtil as ThreadLocal

**Comment by kingster** [comment]
As per the documentation CharsetEncoded is not thread-safe, but I don't think that is the case with CharSet. You are not getting exception because you moved the encoder inside the function causing to create new instance per function call.

**Comment by kingster** [comment]
Charset is thread safe http://docs.oracle.com/javase/1.5.0/docs/api/java/nio/charset/Charset.html

**Comment by ayush03agarwal** [comment]
Added CharsetEncoded in ThreadLocal

File: commons/src/main/scala/com/flipkart/connekt/commons/utils/SmsUtil.scala [hunk 1]
```diff
@@ -12,21 +12,32 @@
...
-import java.nio.charset.Charset
+import java.nio.charset.{Charset, CharsetEncoder}
 
 import com.flipkart.connekt.commons.iomodels.SmsMeta
 import org.smpp.charset.Gsm7BitCharsetProvider
 
 object SmsUtil {
 
   val GSM_CHARSET = new Gsm7BitCharsetProvider().charsetForName("X-Gsm7Bit")
+  val gsmCharsetEncoderPool = new ThreadLocal[CharsetEncoder]
```
Comment (kingster) [inline commons/src/main/scala/com/flipkart/connekt/commons/utils/SmsUtil.scala]: This isn't a pool! I would preferably name this as `localGSMCharsetEncoder` or `gsmCharsetEncoderThreadLocal` or sort of.

Comment (ayush03agarwal) [inline commons/src/main/scala/com/flipkart/connekt/commons/utils/SmsUtil.scala]: changed

**Comment by kingster** [comment]
I have squashed and merged the pr. You would need to delete your branch, and sync changes from flipkart repo, otherwise your future pr will keep adding duplicate commits.

**Pull request merged**